### PR TITLE
Add Delete/Post/Patch functionality to ArticleMasterPage

### PR DESCRIPTION
**Pull request by SebastianStupp**
https://loginicer-pi-addmodal-vitk3joz.herokuapp.com/articlemaster

### CODE REVIEW

File: backend/routes/articles.js [hunk 1]
```diff
@@ -8,15 +8,15 @@ router.get('/', (request, response) => {
 });
 
 router.post('/', (request, response) => {
-  const articleNumber = request.body.articlenumber;
-  const client = request.body.clientname;
-  const bbd = request.body.bdd;
+  const articlenumber = request.body.articlenumber;
```
Comment (lmachens) [inline backend/routes/articles.js:11]: I would revert this change.
`articleNumber` is ;).

Actually, you should call this post request with the correct property names.

File: backend/routes/articles.js [hunk 2]
```diff
@@ -8,15 +8,15 @@ router.get('/', (request, response) => {
...
+  const client = request.body.client;
+  const bbd = request.body.bbd;
   const pzn = request.body.pzn;
   const ean = request.body.ean;
   const time = request.body.time;
```
Comment (lmachens) [inline backend/routes/articles.js:16]:
```suggestion
  const { articleNumber, client, bbd, pzn, ean, time } = request.body;
```

File: backend/routes/articles.js [hunk 3]
```diff
@@ -26,6 +26,30 @@ router.post('/', (request, response) => {
 
   newArticle
     .save()
+    .then((newArticle) => response.json(newArticle))
+    .catch((err) => response.json('Error:' + err));
```
Comment (lmachens) [inline backend/routes/articles.js:30]: Make sure to send an HTTP status code to tell the client, that the request failed.
Same issue in other requests.

```suggestion
    .catch((err) => response.status(500).json('Error:' + err));
```

File: client/src/components/AddModal.js [hunk 4]
```diff
@@ -43,52 +38,149 @@ const Modal = styled.div`
...
+      if (location.pathname === '/articlemaster') {
+        setModalTypeArticle(!modalTypeArticle);
+      }
+
+      if (location.pathname === '/clientmaster') {
+        setModalTypeClient(!modalTypeClient);
+      }
+    }, []);
+  }
+  DeclareModalType();
```
Comment (lmachens) [inline client/src/components/AddModal.js:100]: This is weired :). 

```suggestion
    React.useEffect(() => {
      if (location.pathname === '/articlemaster') {
        setModalTypeArticle(!modalTypeArticle);
      }

      if (location.pathname === '/clientmaster') {
        setModalTypeClient(!modalTypeClient);
      }
    }, []);
  
```

File: client/src/components/AddModal.js [hunk 5]
```diff
@@ -43,52 +38,149 @@ const Modal = styled.div`
...
+    await doPostArticle(articleNumber, client, bbd, pzn, ean);
+    close();
+  }
+
+  function handleClientChange(client) {
+    setClient(client);
+  }
   return (
     <>
+      {loadingArticle}
```
Comment (lmachens) [inline client/src/components/AddModal.js:117]: Is `loadingArticle` a boolean?
This will not render anything.

File: client/src/components/AddModal.js [hunk 6]
```diff
@@ -43,52 +38,149 @@ const Modal = styled.div`
...
 
-export default function DeleteModal({ close }) {
+const CheckboxContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-around;
+  padding: 20px;
+`;
+
+export default function AddModal({ close }) {
```
Comment (lmachens) [inline client/src/components/AddModal.js:76]: Components and HTML elements accept events. We should do the same in our custom Components.
For you, it's a simple naming change (might be the same issue in other components).

```suggestion
export default function AddModal({ onClose }) {
```

File: client/src/components/ChangeModal.js [hunk 7]
```diff
@@ -43,49 +38,162 @@ const Modal = styled.div`
...
+                  setPznValue(!pznValue);
+                }}
+              ></Checkbox>
+              <Checkbox
+                checkStatus={ean ? 'checked' : null}
+                option="EAN Number"
+                checked={() => {
+                  setEanValue(!eanValue);
+                }}
+              ></Checkbox>
```
Comment (lmachens) [inline client/src/components/ChangeModal.js:171]: I know checkbox from HTML:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input/checkbox
https://reactjs.org/docs/forms.html#handling-multiple-inputs

I can use it like this:
`<input type="checkbox" checked={checked} onChange={(event) => setChecked(event.target.checked)} >`

Your checkbox is different, which might be confusing for other developers.

1: `checked` is equal to `onChange`
2: `checkedStatus` is equal to `checked`

Better not change the meaning of these properties:

```suggestion
              <Checkbox
                checked={ean}
                option="EAN Number"
                onChange={(event) => {
                  setEanValue(event.target.checked);
                }}
              />
```

You will reduce the complexity of your code.

File: client/src/components/DeleteModal.js [hunk 8]
```diff
@@ -53,25 +55,67 @@ const Modal = styled.div`
...
+
+      if (location.pathname === '/clientmaster') {
+        setModalTypeClient(!modalTypeClient);
+      }
+    }, []);
+  }
+  DeclareModalType();
 
-  async function handleClick() {
+  async function handleonClickClient() {
```
Comment (lmachens) [inline client/src/components/DeleteModal.js:85]: 😄

File: client/src/hooks/usePatchArticle.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  ) {
+    try {
+      const article = await patchArticle(
+        articleId,
+        articlenumber,
+        client,
+        bbd,
+        pzn,
+        ean
+      );
```
Comment (lmachens) [inline client/src/hooks/usePatchArticle.js:25]: You could save some lines of code if you change your functions to accept only one or two arguments:

```suggestion
  async function doPatchArticle(
    articleId,
    article
  ) {
    try {
      const article = await patchArticle(
        articleId,
        article
      );
```

or

```suggestion
  async function doPatchArticle(
    article
  ) {
    try {
      const article = await patchArticle(
        articleId,
      );
```

**Comment by SebastianStupp** [comment]
Thx for the input, as soon as the basic functionality is ready, I will refactor some code
